app/intune/index.js

```diff
@@ -0,0 +1,91 @@
+const dbus = require('@homebridge/dbus-native');
+const { LucidLog } = require('lucid-log');
+
+var sessionBus = dbus.sessionBus();
+
+var intuneAccount = null;
+
+var brokerService = sessionBus.getService('com.microsoft.identity.broker1');
+
+function processInTuneAccounts(logger, resp, ssoInTuneAuthUser) {
+	response = JSON.parse(resp);
+	if ('error' in response) {
+		logger.warn('Failed to retrieve InTune account list: ' + response.error.context);
+		return;
+	};
+
+	if (ssoInTuneAuthUser == '') {
+		intuneAccount = response.accounts[0];
+		logger.debug('Using first available InTune account (' + intuneAccount.username + ')');
+	} else {
+		for (account in response.accounts) {
+			if (account.username == ssoIntuneAuthUser) {
+				intuneAccount = account;
+				logger.debug('Found matching InTune account (' + intuneAccount.username + ')');
+				break;
+			}
+		}
+		if (intuneAccount == null) {
+			logger.warn('Failed to find matching InTune account for ' + ssoIntuneAuthUser + '.');
+		}
+	}
+}
+
+exports.initSso = function initIntuneSso(logger, ssoInTuneAuthUser) {
+	logger.debug("Initializing InTune SSO");
+	brokerService.getInterface(
+		'/com/microsoft/identity/broker1',
+		'com.microsoft.identity.Broker1', function(err, broker) {
+			if (err) {
+				logger.warn('Failed to find microsoft-identity-broker DBus interface');
+				return;
+			}
+			broker.getAccounts('0.0', '', JSON.stringify({'clientId': '88200948-af09-45a1-9c03-53cdcc75c183', 'redirectUri':'urn:ietf:oob'}), function(err, resp) {
```

Comment:
Interestingly you get a different response from the broker if you use a specific client id (like the one from Edge, `d7b530a4-7680-4c23-a8bf-c52c121d2e87`). When using that, I just get a single account back instead of three time the same.

Reply:
That single account is highly likely the MS Entra identity for the synchronized work profile in MS Edge. I'm not sure how this is going to react in case you have a second work profile synchronized to a different MS Entra user identity.

Reply:
But I already experienced that MS Intune had issues in the past in that multiple work profiles scenario.